client/core/bond.go

```diff
@@ -39,58 +40,217 @@ func (c *Core) watchExpiredBonds(ctx context.Context) {
+//
+//    E = t3 - t2, *constant* duration, dex.BondExpiry()
+//    p = t1 - t0, *variable*, a random process
+//    m = t4 - t3, *variable*, depends on consensus rules and blocks
+//        e.g. several blocks after lockTime passes
+//
+//  - bonds may be spent at t4
+//  - bonds must be replaced by t2 i.e. broadcast by some t2'
+//  - perfectly aligning t4 for bond A with t2' for bond B is impossible on
+//    account of the variable durations
+//  - t2-t2' should be greater than a large percent of expected pending
+//    durations (see pendingBuffer)
+//
+// Here a replacement bond B had a long pending period, and it became active
+// after bond A expired (too late):
+//
+//             t0  t1                         t2' t2                   t3
+//   bond A:   |~~~|--------------------------^---|====================|
+//                                                x
+//   bond B:                                  |~~~~~~|------------------...
+//
+// Here the replacement bond was confirmed quickly, but l was too short,
+// causing it to expire before bond A became spendable:
+//                                                                        > renew as bond C
+//   bond A:   |~~~|----------------------^-------|====================|++‖~~~~~|---...
+//                                          ✓                             x     x
+//   bond B:                              |~|------------------------|=====...
+//
+// Similarly, l could have been long enough to broadcast a replacement in time,
+// but the pending period could be too long (the second "x").
+//
+// Here the replacement bond was broadcast with enough time to confirm before
+// the previous bond expired, and the previous bond became spendable in time to
+// broadcast and confirm another replacement (sustainable):
+//                                                                        > renew as bond C
+//   bond A:   |~~~|----------------------^-------|====================|++‖~~~~~|---...
+//                                               ✓                        ✓     ✓
+//   bond B:                              |~~~~~~|-------------------------------|====...
+//
+// Thus, bond rotation without tier drops requires l>E+m+p. For
+// t3-t0 = p+l+E, this means t3-t0 >= 2(E+p)+m. We will assume the time
+// from authoring to broadcast is negligible, absorbed into the estimate of the
+// max pending duration.
+//
+// tldr:
+//  - when creating a bond, set lockTime = 2*(BondExpiry+pendingBuffer)+spendableDelay
+//  - create a replacement bond at lockTime-BondExpiry-pendingBuffer
+
+// pendingBuffer gives the duration in seconds prior to reaching bond expiry
+// (account, not lockTime) after which a new bond should be posted to avoid
+// account tier falling below target while the replacement bond is pending. The
+// network is a parameter because expected block times are network-dependent,
+// and we require short bond lifetimes on simnet.
```

Comment:
> Instead of a universal approximation, why not have the `Bonder` specify generous estimates for the asset-specific variables`p` and `m` and use `2p + m` (I think) from above?

Right, I've alluded to this idea in the docs for `spendableDelay` below.  

```
// .... NOTE: to minimize bond overlap, an asset.Bonder method could provide
// this estimate, but it is still very short relative to the entire bond
// lifetime, which is on the order of months.
```

I'm just convincing myself that it would matter enough to justify the API bloat and additional code.  Not yet seeing it, but may yet do this when the reserves is working right.

> To clarify, pendingBuffer is not an approximation for pending (p) = t1 - t0, it's t2 - t2'?

That's correct.  This pertains to the second "tldr" bullet from the giant block comment above:

```
// tldr:
//  - when creating a bond, set lockTime = 2*(BondExpiry+pendingBuffer)+spendableDelay
// - create a replacement bond at lockTime-BondExpiry-pendingBuffer
```